docs/pages/management/export-audit-events/datadog.mdx

```diff
@@ -0,0 +1,417 @@
+key = "/home/sasha/scripts/event-handler/client.key"
+url = "https://localhost:8888/test.log"
+
+[teleport]
+addr = "mytenant.teleport.sh:443"
+identity = "identity"
+```
+
+To start the event handler, run the following command:
+
+```code
+$ ./teleport-event-handler start --config teleport-event-handler.toml
+```
+
+</TabItem>
+<TabItem scope={["oss", "enterprise"]} label="Self-Hosted">
+
+```toml
+storage = "./storage"
+timeout = "10s"
+batch = 20
+namespace = "default"
+
+[forward.fluentd]
+ca = "/home/sasha/scripts/event-handler/ca.crt"
+cert = "/home/sasha/scripts/event-handler/client.crt"
+key = "/home/sasha/scripts/event-handler/client.key"
+url = "https://localhost:8888/test.log"
+
+[teleport]
+addr = "teleport.example.com:443"
+identity = "identity"
+```
+
+To start the event handler, run the following command:
+
+```code
+$ ./teleport-event-handler start --config teleport-event-handler.toml
+```
+
+</TabItem>
+<TabItem scope={["oss", "enterprise"]} label="Helm Chart">
+
+Use the following template to create `teleport-plugin-event-handler-values.yaml`:
+
+```yaml
+eventHandler:
+  storagePath: "./storage"
+  timeout: "10s"
+  batch: 20
+  namespace: "default"
+
+teleport:
+  address: "example.teleport.com:443"
+  identitySecretName: teleport-event-handler-identity
+
+fluentd:
+  url: "https://fluentd.fluentd.svc.cluster.local/events.log"
+  sessionUrl: "https://fluentd.fluentd.svc.cluster.local/session.log"
+  certificate:
+    secretName: "teleport-event-handler-client-tls"
+    caPath: "ca.crt"
+    certPath: "client.crt"
+    keyPath: "client.key"
+
+persistentVolumeClaim:
+  enabled: true
+```
+
+To start the event handler in Kubernetes, run the following command:
+
+```code
+$ helm install teleport-plugin-event-handler teleport/teleport-plugin-event-handler \
+  --values teleport-plugin-event-handler-values.yaml \
+  --version (=teleport.plugin.version=)
+```
+
+</TabItem>
+</Tabs>
+
+<Admonition type="note">
+
+This example will start exporting from `May 5th 2021`:
+
+```code
+$ ./teleport-event-handler start --config teleport-event-handler.toml --start-time "2022-02-02T00:00:00Z"
+```
+
+The start time can be set only once, on the first run of the tool.
+
+If you want to change the time frame later, remove the plugin state directory
+that you specified in the `storage` field of the handler's configuration file.
+
+</Admonition>
+
+Once the handler starts, you will see notifications in Fluentd about scanned and forwarded events:
+
+```txt
+INFO[0046] Event sent id=0b5f2a3e-faa5-4d77-ab6e-362bca0994fc ts="2021-06-08 11:00:56.034 +0000 UTC" type=user.login
+...
+```
+
+The Logs view in Datadog should now report your Teleport cluster events:
+
+![Datadog Logs](../../../img/management/datadog-logs.png)
+
+Review the Fluentd output plugin for Datadog [README
+file](https://github.com/DataDog/fluent-plugin-datadog/blob/master/README.md)
+to learn how to customize the log format entering Datadog.
+
+## Troubleshooting connection issues
+
```

Comment:
I got "connection reset by peer" errors when running the Event Handler:

```
ERRO   Error sending event to Teleport: Post "https://localhost:8888/test.log": read tcp 127.0.0.1:39460->127.0.0.1:8888: read: connection reset by peer event-handler/app.go:118
```

Looks like this has to do with the TLS handshake between the Event Handler and Fluentd, as reproduced via `curl`:

```
$ curl --key client.key --cert client.crt --cacert ca.crt -v https://localhost:8888/test.log
*   Trying 127.0.0.1:8888...
* TCP_NODELAY set
* Connected to localhost (127.0.0.1) port 8888 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* successfully set certificate verify locations:
*   CAfile: ca.crt
  CApath: /etc/ssl/certs
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
* OpenSSL SSL_connect: Connection reset by peer in connection to localhost:8888
* Closing connection 0
curl: (35) OpenSSL SSL_connect: Connection reset by peer in connection to localhost:8888
```

Weirdly, the client cert's `Not After` is before its `Not Before`:

```
~$ openssl x509 -text -in client.crt
Certificate:
    Data:
        Version: 3 (0x2)
        Serial Number:
            8a:82:e4:d5:f1:7b:c4:e8:a9:56:8e:ee:64:7f:87:7c
        Signature Algorithm: sha256WithRSAEncryption
        Issuer: C = US, CN = localhost
        Validity
            Not Before: Feb 23 23:22:46 2023 GMT
            Not After : Feb 20 23:22:46 2033 GMT
```

I built the Event Handler from commit `bc85392dc3b03bedcd963709d3ef4bba2af37b2d`.

Reply:
That's from [this issue](https://github.com/gravitational/teleport-plugins/issues/747). I'm hoping it will be resolved soon, since it affects all guides using the generated configuration.